src/modules/authorizer/role/AUT_TokenGated_Roles_v1.sol

```diff
@@ -89,42 +89,89 @@ contract AUT_TokenGated_Roles_v1 is IAUT_TokenGated_Roles_v1, AUT_Roles_v1 {
     // Stores the threshold amount for each token in a role
     mapping(bytes32 => uint) public thresholdMap;
 
+    // Storage gap for future upgrades
+    uint[50] private __gap;
+
     //--------------------------------------------------------------------------
     // Overloaded and overriden functions
 
-    function hasRole(bytes32 roleId, address account)
+    function hasRole(bytes32 roleId, address who)
         public
         view
         virtual
         override(AccessControlUpgradeable, IAccessControl)
         returns (bool)
     {
         if (isTokenGated[roleId]) {
-            return _hasTokenRole(roleId, account);
+            return _hasTokenRole(roleId, who);
```

Comment:
@0xNuggan

Reply:
yes, hasRole will only return true if the balance is above the threshold, and that means there is no direct way to check if a token has a role or not (as you said, the way would be to go through the threshold). We can add that, it's a clear QoL improvement. On the other point, that's why we enforce there being a threshold that late in the process, so it regverts if somebody tries to set a tokengated role to a user. Keep in mind that token-gated and regular roles can coexist in a single authorizer

Reply:
My point is here that you are breaking expected behavior, as `hasRole(role, account)` does not mean anymore that the account has that role. 
This is bound to give you problems at the certain point, because other functions may expect that hasRole means what it says on the surface. For example, _revokeRole and _grantRole  in the OZ modules do, and wil now not anymore behave as you would expect.